src/zip.ts

```diff
@@ -0,0 +1,117 @@
+import fs from 'fs';
+import path from 'path';
+import { execSync } from 'child_process';
+import { platform } from 'os';
+import { errCode } from './error';
+
+function validate(workspace: string, source: string, dest: string) {
+  if (!source.trim()) {
+    throw new Error('The source path cannot be empty');
+  }
+  if (!dest.trim()) {
+    throw new Error('The destination file cannot be empty');
+  }
+
+  // Verify the source folder exists and is not a file.
+  try {
+    const stats = fs.statSync(source);
+    if (!stats.isDirectory()) {
+      throw new Error('Invalid source folder: the source must be a directory');
+    }
+  } catch (err) {
+    if (errCode(err) === 'ENOENT') {
+      throw new Error('Invalid source folder: not exist');
+    }
+  }
+
+  if (path.isAbsolute(source)) {
+    throw new Error('Invalid source folder: absolute path is not supported');
+  }
+  if (isFolderOutside(source, workspace)) {
+    throw new Error(
+      'Invalid source folder: the source path is outside of the user workspace',
+    );
+  }
+  if (!dest.endsWith('.zip')) {
+    throw new Error('Invalid zip filename: Only .zip file is permitted');
+  }
+}
+
+/**
+ * Checks if a folder is outside of a specified folder.
+ *
+ * Contextual note: Ordinarily, a folder cannot be considered outside of itself.
+ * However, in scenarios where the targetFolder equals the specifiedFolder,
+ * it implies an attempt to archive the entire workspace.
+ * Such actions are prohibited, thus leading to a return value of true.
+ *
+ * @param {string} targetFolder The path to the target folder.
+ * @param {string} specifiedFolder The path to the specified folder.
+ * @returns {boolean} Returns true if the target folder is outside of the specified folder, false otherwise.
+ */
+export function isFolderOutside(
+  targetFolder: string,
+  specifiedFolder: string,
+): boolean {
```

Comment:
I find the code a bit hard to read when they refer to the vars `targetFolder` and `specifiedFolder`. The names are not conclusive enough to let me understand what is what without looking up the description again and again. How about `rootFolder` and `subFolder`?

Reply:
👍 Better!